doc/tutorials/ostro-oic-tutorial.md

```diff
@@ -0,0 +1,1030 @@
+'state' and 'power'. One can think of a light, for instance. It's either
+'ON' or 'OFF' - the state, and has different light intensities - the power.
+We can use two accumulators for this task: one from 0 to 1, to store
+'state', and another from 0 to 100 to store 'power'.
+
+```
+power_value(int/accumulator:setup_value=min:0|max:100|step:1,initial_value=50)
+state_value_as_int(int/accumulator:setup_value=min:0|max:1)
+```
+
+As `int/accumulator` node type deals with integers, and 'state' is more
+naturally represented as boolean, we will need some 'converters' to change
+from/to integer/boolean:
+
+```
+state_value_from_boolean(converter/boolean-to-int) OUT -> SET state_value_as_int
+state_value_as_int OUT -> IN state_value(converter/int-to-boolean)
+```
+
+The first converter, `state_value_from_boolean`, will receive boolean
+packets and converter to integer and deliver the result to
+`state_value_as_int`. The second one does the opposite: it receives the
+output from `state_value_as_int` and converts to boolean.
+
+### Running the application
+
+If we write those to a file named 'controller.fbp', we can run it by:
+
+```
+$ sol-fbp-runner controller.fbp
+```
+
+The result is that nothing will appear. As Soletta applications are
+intended to run on devices, they should run forever. And as our
+application has no input or output, it will just block our terminal.
+You can finish it with ^C (Ctrl+C).
+
+## Step 2
+
+### Seeing initial values
+
+So far, our application was not very useful. It did nothing! Now, let's
+add some interaction, so we can see if 'state' and 'power' behave as
+expected.
+To debug it, we can always send nodes output to `console` node type:
+
+```
+power_value OUT -> IN power_value_output(console)
+state_value OUT -> IN state_value_output(console)
+```
+
+If we run this updated application, we shall see:
+
+```
+power_value_output 50 (integer range)
+state_value_output false (boolean)
+```
+
+This is so because many nodes, when 'are born', send they initial value
+through its ports, so we basically see they initial values.
+
+### Changing some states
+
+Now, let's start interacting with our app. We'll use three buttons,
+one to toggle 'state' value, one to increase 'power' and another to
+decrease it. At this first moment, we'll use desktop keyboard buttons.
+
+```
+increase_button(keyboard/boolean:binary_code=106)
+decrease_button(keyboard/boolean:binary_code=107)
+state_button(keyboard/boolean:binary_code=97)
+```
+
+Those binary codes correspond to 'j', 'k' and 'a', respectively.
+It's time to connect them to their accumulators. As buttons outputs
+a `true` packet when pressed and `false` when released, we need to
+filter its output: only when they are pressed matters.
+We achieve this with node type `boolean/filter` - connect only its
+TRUE output port to get only true packets.
+
+```
+increase_button OUT -> IN _(boolean/filter) TRUE -> INC power_value
```

Comment:
nice, I always forget about `boolean/filter`, could you update the oic-and-http-light to reflect this as well?